master/addtrigger.sql

```diff
@@ -21,18 +21,20 @@ FOR EACH ROW EXECUTE PROCEDURE stream_trial_seq_modify();
 
 -- helper function to create trial jsonb object for streaming
 CREATE OR REPLACE FUNCTION stream_trial_notify(
-    before jsonb, beforeproj integer, after jsonb, afterproj integer
+    before jsonb, beforeproj integer, beforework integer, after jsonb, afterproj integer, afterwork integer
 ) RETURNS integer AS $$
 DECLARE
     output jsonb = NULL;
     temp jsonb = NULL;
 BEGIN
     IF before IS NOT NULL THEN
         temp = before || jsonb_object_agg('project_id', beforeproj);
```

Comment:
What is project being streamed for?  It feels like it isn't part of the Trials table, and it isn't part of rbac, so it probably shouldn't be passed through the NOTIFY channel at all anymore

Reply:
I was under the impression that the project_id's were being streamed in case someone was subscribed to all the trials that fell under a specific project? `beforeproj` and `afterproj` had already been in place prior to the PR.

This commit is adds `beforework` and `afterwork` for RBAC/permission scope. Are we looking to drop `beforeproj` and `afterproj`?

Reply:
The APIs haven't been written yet, but I want us to take a conservative approach to the API we expose.  It's easy to add API surface area later, it's hard to remove it.

I think we strive to stream what's in each table and nothing more.  I think if somebody wants all trials in a project, they need to subscribe to all experiments in a project (project is a column in experiments) then subscribe to all trials in each of those experiments.

We can increase complexity of the master as we see client use-cases pop up where this sort of strategy creates undue burden.  But realistically, if you want to write a streaming client, you probably need to expect some amount of burden... streaming is a complex feature.

Additionally, "stream only what is in the tables" is exactly what we agreed to in the [stakeholder meeting around the streaming updates proposal](https://hpe-my.sharepoint.com/:w:/p/ryan_beethe/EXdsETYELNpPnRkTJ8s3qFcBXS0WSWXJAZmQLzipaMaZhw?e=E1uiLx).  So nobody can complain if that's the limitation we start with.

Reply:
(so yes, remove them.  I don't see what purpose beforeproj and afterproj serve, if we're aiming to ship just what's in the trials table)